rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/utils/HttpUtils.java

```diff
@@ -475,6 +475,9 @@ public static String getProtocolHeader(Message m, String name, String defaultVal
 
     public static String getBaseAddress(Message m) {
         String endpointAddress = getEndpointAddress(m);
+        if (endpointAddress.startsWith("ws")) {
```

Comment:
Curios why we substitute `ws` with `http`? I believe the call will fail (plus we also could have `wss://`)

Reply:
This change is required to fix some websocket related tests failures with Jetty 12.

We need this change because in Jetty 12 architect and API changes.
For example, if the request url is "ws://localhost:31604/websocket" or "http://localhost:31604/websocket", we want getBaseAddress() returns "/websocket".

I pushed another commit here to honor "wss" protocol as well
https://github.com/apache/cxf/pull/1633/commits/20a0f1c5230c4907b2b03db1a7a12d7a0024baaf